metadata/src/main/java/org/apache/kafka/image/writer/ImageReWriter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.image.writer;
+
+import org.apache.kafka.image.ImageProvenance;
+import org.apache.kafka.image.MetadataDelta;
+import org.apache.kafka.image.MetadataImage;
+import org.apache.kafka.server.common.ApiMessageAndVersion;
+
+
+/**
+ * ImageReWriter writes a metadata image out to another metadata image.
+ *
+ * There are a few reasons why you might want to do this. One is to obtain a MetadataDelta
+ * object which contains everything in the image. Another is to translate an image from
+ * one metadata version to another.
+ */
+public class ImageReWriter implements ImageWriter {
+    private final String newSource;
+    private final MetadataDelta delta;
+    private boolean closed = false;
+    private MetadataImage image = null;
+
+    public ImageReWriter(
+        String newSource,
+        MetadataDelta delta
+    ) {
```

Comment:
Hmm. Maybe I missed it but doesn't look like this class is used anywhere.

I was interested in its usage because it wasn't clear to me when it would be useful to pass in the delta. Instead of starting with a empty delta from an empty or non-empty image.